target/scripts/helpers/relay.sh

```diff
@@ -25,73 +74,112 @@ function _relayhost_sasl
   then
     _log 'trace' "Adding relay authentication from postfix-sasl-password.cf"
 
-    # add domain-specific auth from config file:
+    # Add domain-specific auth from config file:
     while read -r LINE
     do
       if ! _is_comment "${LINE}"
       then
         echo "${LINE}" >> /etc/postfix/sasl_passwd
       fi
     done < /tmp/docker-mailserver/postfix-sasl-password.cf
+
+    # Only relevant when providing this user config (unless users append elsewhere too)
+    postconf 'smtp_sender_dependent_authentication = yes'
   fi
 
-  # add default relay
+  # Add an authenticated relay host defined via ENV config:
   if [[ -n ${RELAY_USER} ]] && [[ -n ${RELAY_PASSWORD} ]]
   then
-    # white-space separates value pairs (any length is valid)
-    echo "[${RELAY_HOST}]:${RELAY_PORT} ${RELAY_USER}:${RELAY_PASSWORD}" >> /etc/postfix/sasl_passwd
+    echo "$(_env_relay_host)    ${RELAY_USER}:${RELAY_PASSWORD}" >> /etc/postfix/sasl_passwd
   fi
 
   _sasl_set_passwd_permissions
+
+  # Technically if only a single relay host is configured, a `static` lookup table could be used instead?:
+  # postconf "smtp_sasl_password_maps = static:${RELAY_USER}:${RELAY_PASSWORD}"
+  postconf 'smtp_sasl_password_maps = texthash:/etc/postfix/sasl_passwd'
```

Comment:
I like the `texthash` method :)

Reply:
In future it may be better to use `hash` table type or provide a way to support both. This would only be relevant if the change detection was smarter on what changes were applied.

At least it should be better in situations where some databases are updated at a higher frequency, eg on a larger scale via API. For bulk updates it's probably not a big issue as long as the service disruption is minimal.